packages/devtools_app/lib/src/screens/app_size/app_size_controller.dart

```diff
@@ -160,21 +174,39 @@ class AppSizeController {
     }
   }
 
+  DiffTreeMap? get _activeDiffMap {
+    final appUnit = _selectedAppUnit.value;
+
+    switch (appUnit) {
+      case AppUnit.mainOnly:
+        return _mainDiffTreeMap;
+      case AppUnit.deferredOnly:
+        return _deferredDiffTreeMap;
+      case AppUnit.entireApp:
+      default:
+        return _diffTreeMap;
+    }
+  }
+
   TreemapNode? get _activeDiffRoot {
-    switch (_activeDiffTreeType.value) {
+    final diffTreeType = _activeDiffTreeType.value;
+    final diffMap = _activeDiffMap;
```

Comment:
Let's return `null` if `_activeDiffMap` is `null` (instead of null-asserting with `!`)